woocommerce/woocommerce
# Adding ProductForm data store

### All Submissions:


### Changes proposed in this Pull Request:

Adding the JS data store for accessing the Product Form API data.

Closes #36073  .




### How to test the changes in this Pull Request:



**Note: you may have to run `composer dump-autoload` for the new classes to show.
1. In your `mu-plugin` add this action:
```php
/**
 * Add Example items to WooCommerce Navigation
 */
function add_product_form_field() {
    if (
        ! method_exists( '\Automattic\WooCommerce\Internal\Admin\ProductForm\FormFactory', 'add_field' )
    ) {
        return;
    }

    \Automattic\WooCommerce\Internal\Admin\ProductForm\FormFactory::add_field(
        'test_new_field',
        'woocommerce',
        array(
            'type'       => 'text',
            'section'    => 'product-details',
            'properties' => array(
                'name'       => 'name',
                'label'      => 'label',
            )
        )
    );
}
add_action( 'init', 'add_product_form_field' );
```
2. Enable the `new-product-management-experience` feature flag`
3. Refresh, open the console and enter `wp.data.select( 'experimental/wc/admin/product-form' ).getFields()`
**Note: you may have to run this twice to see the field show up.
You should see something that looks like:
```js
[
  {
    "id": "test_new_field",
    "plugin_id": "woocommerce",
    "order": 20,
    "type": "text",
    "section": "product-details",
    "properties": {
      "name": "name",
      "label": "label"
    }
  }
]
```
4. Enter `wp.data.select( 'experimental/wc/admin/product-form' ).getProductForm()` into the console.
You should see:
```js
{
  "fields": [ { /*...same as above object */ } ],
  "sections": [],
  "subsections": []
}
```



### Other information:




### FOR PR REVIEWER ONLY:
